internal/resources-validation.yml

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2023 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+jobs:
+# Spin up a Linux VM for validating code.
+- job: png_resources_validation
+  displayName: Resources Validation
+  pool:
+    vmImage: 'ubuntu-latest'
+
+  steps:
+    - script: |
+        if [ ! -d "$(Build.SourcesDirectory)/resources" ]; then
+          echo "No resources directory found, skipping job."
+          exit 0
+        fi
+      displayName: 'Check for resources'
+    - script: |
+        sudo apt update
+        sudo apt install -y build-essential
+        wget https://downloads.sourceforge.net/project/pmt/pngcrush/${{ parameters.pngcrush_ref }}/pngcrush-${{ parameters.pngcrush_ref }}.tar.gz
+        tar xzf pngcrush-${{ parameters.pngcrush_ref }}.tar.gz
+        cd pngcrush-${{ parameters.pngcrush_ref }}
+        make
```

Comment:
Wow, we compile the product every time now. I am sorry I made you do that.

Another way to make sure this "string parsing" method always works is to store 2 PNG files in this very repos: 1 valid and 1 invalid. And to test them with the same script during CI. That way, we don't have to compile and to stick to a specific version.

Reply:
Thanks, guys. I will definitely check that out

Reply:
@carlos-villavicencio-adsk yes, this is what Norberto was doing before my comment: https://github.com/shotgunsoftware/tk-ci-tools/pull/46#discussion_r1403647199

But now I suggest the good image/bad image test approach for each test, we should be able to go back to a simple apt install.